streaming/src/main/scala/org/apache/spark/streaming/DStreamGraph.scala

```diff
@@ -31,12 +31,15 @@ final private[streaming] class DStreamGraph extends Serializable with Logging {
   private val inputStreams = new ArrayBuffer[InputDStream[_]]()
   private val outputStreams = new ArrayBuffer[DStream[_]]()
 
+  val inputStreamNameAndID = new ArrayBuffer[(String, Int)]()
```

Comment:
nit: change it to `@volatile private var inputStreamNameAndID: Seq[(String, Int)] = Nil` and just set it in `start`. Don't expose a mutable ArrayBuffer to the caller.